bootstrap/pkg/kfapp/ksonnet/ksonnet.go

```diff
@@ -162,14 +177,21 @@ func (ksApp *ksApp) applyComponent(components []string, cfg *clientcmdapi.Config
 		if len(doneApply) == len(components) {
 			return nil
 		}
-		return fmt.Errorf("%v failed components in last try", len(components)-len(doneApply))
+		return &kfapis.KfError{
+			Code:    int(kfapis.INTERNAL_ERROR),
+			Message: fmt.Sprintf("%v failed components in last try", len(components)-len(doneApply)),
+		}
 	}, bo)
 	if err != nil {
 		log.Errorf("components apply failed; Error: %v", err)
+		return &kfapis.KfError{
+			Code:    int(kfapis.INTERNAL_ERROR),
+			Message: err.Error(),
```

Comment:
Can we keep error message:
("components apply failed; Error: %v"...

Reply:
done.